File: hello-world/clients/unity/Assets/Scripts/socket/SocketProxy.cs
using com.tvd12.ezyfoxserver.client;
using com.tvd12.ezyfoxserver.client.config;
using com.tvd12.ezyfoxserver.client.evt;
using com.tvd12.ezyfoxserver.client.constant;
using com.tvd12.ezyfoxserver.client.handler;
using com.tvd12.ezyfoxserver.client.request;
using com.tvd12.ezyfoxserver.client.util;
using UnityEngine;

class DisconnectionHandler : EzyDisconnectionHandler
{
    protected override void postHandle(EzyDisconnectionEvent evt)
    {
        logger.info("event = " + evt);
    }
}

class HandshakeHandler : EzyHandshakeHandler
{
    protected override EzyRequest getLoginRequest()
    {
        return new EzyLoginRequest("example", "username", "passworld");
    }
}

public class SocketProxy : EzyLoggable
{
    private static readonly SocketProxy INSTANCE = new SocketProxy();

    public const string ZONE_NAME = "hello-world";
    public const string PLUGIN_NAME = "hello-world";

    private EzyClient client;
    private string host;
    private int port;

    public EzyClient Client { get => client; }
    public string Host { get => host; }
    public int Port { get => port; }

    public static SocketProxy getInstance()
    {
        return INSTANCE;
    }

    public EzyClient setup(string host, int port)
    {
        this.host = host;
        this.port = port;

        logger.debug("Set up socket client");
        var config = EzyClientConfig.builder()
            .clientName(ZONE_NAME)
            .pingConfigBuilder()
                .pingPeriod(1000)
                .maxLostPingCount(5)
                .done()
            .build();

        var clients = EzyClients.getInstance();
        client = clients.newDefaultClient(config);
        var setup = client.setup();
        setup.addEventHandler(EzyEventType.DISCONNECTION, new DisconnectionHandler());
        setup.addDataHandler(EzyCommand.HANDSHAKE, new HandshakeHandler());

        return client;
    }

    public void connect()
    {
        client.connect("tvd12.com", 3005);
    }

}

File: swift-connection/src/SocketProxy.swift
//
//  SocketProxy.swift
//  freechat-swift
//
//  Created by Dzung on 10/05/2021.
//  Copyright © 2021 Young Monkeys. All rights reserved.
//

import Foundation

public let ZONE_NAME = "example"
public let APP_NAME = "hello-world"

public class SocketProxy {
    
    private let client: EzyClient
    private static let INSTANCE = SocketProxy()
    
    private init() {
        let config = EzyClientConfig()
            .setClientName(clientName: ZONE_NAME)
            .setZoneName(zoneName: ZONE_NAME)
            .setEnableSSL()
//            .setEnableDebug()
        _ = config.ping
            .setPingPeriod(pingPeriod: 3000)
            .setMaxLostPingCount(maxLostPingCount: 3)
        _ = config.reconnect
            .setReconnectPeriod(reconnectPeriod: 1000)
            .setMaxReconnectCount(maxReconnectCount: 10)
        let clients = EzyClients.getInstance()!
        client = clients.newClient(config: config)
        let setup = client.setup!
            .addEventHandler(eventType: EzyEventType.CONNECTION_SUCCESS, handler: ExConnectionSuccessHandler())
            .addEventHandler(eventType: EzyEventType.CONNECTION_FAILURE, handler: EzyConnectionFailureHandler())
            .addEventHandler(eventType: EzyEventType.DISCONNECTION, handler: ExDisconnectionHandler())
            .addDataHandler(cmd: EzyCommand.LOGIN, handler: ExLoginSuccessHandler())
            .addDataHandler(cmd: EzyCommand.APP_ACCESS, handler: ExAppAccessHandler())
            .addDataHandler(cmd: EzyCommand.HANDSHAKE, handler: ExHandshakeHandler())
        _ = setup.setupApp(appName: APP_NAME)
        Thread.current.name = "main";
        clients.processEvents()
    }
    
    public static func getInstance() -> SocketProxy {
        return INSTANCE
    }
    
    public func connectToServer() {
        let host = "127.0.0.1"
//        let host = "ws.tvd12.com"
        client.connect(host: host, port: 3005)
    }
}

class ExConnectionSuccessHandler: EzyConnectionSuccessHandler {
}

class ExDisconnectionHandler: EzyDisconnectionHandler {
    override func postHandle(event: NSDictionary) {
        // do something here
    }
}

class ExHandshakeHandler : EzyHandshakeHandler {
    override func getLoginRequest() -> NSArray {
        let array = NSMutableArray()
        array.add(ZONE_NAME)
        array.add("swift123")
        array.add("12345678912")
        return array
    }
};

class ExLoginSuccessHandler : EzyLoginSuccessHandler {
    override func handleLoginSuccess(responseData: NSObject) {
        let array = NSMutableArray()
        array.add(APP_NAME)
        array.add(NSDictionary())
        client!.send(cmd: EzyCommand.APP_ACCESS, data: array)
    }
};

class ExAppAccessHandler : EzyAppAccessHandler {
    override func postHandle(app: EzyApp, data: NSObject) -> Void {
        let requestData = NSMutableDictionary()
        requestData["who"] = "Swift"
        app.send(cmd: "secureChat", data: requestData, encrypted: true)
    }
};
